core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -95,28 +93,70 @@ private static StaticDataTask.Row convertPartition(Partition partition) {
         partition.key, partition.specId, partition.dataRecordCount, partition.dataFileCount);
   }
 
-  private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
-    CloseableIterable<FileScanTask> tasks = planFiles(scan);
+  @VisibleForTesting
+  static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
     Types.StructType normalizedPartitionType = Partitioning.partitionType(table);
     PartitionMap partitions = new PartitionMap();
 
     // cache a position map needed by each partition spec to normalize partitions to final schema
     Map<Integer, int[]> normalizedPositionsBySpec =
         Maps.newHashMapWithExpectedSize(table.specs().size());
+    // logic to handle the partition evolution
```

Comment:
Nit: can we add one line before this?

And also, if we keep comment, we can just do something shorter like : ```handle partition evolution```